core/src/test/resources/sql/set-op.iq

```diff
@@ -258,4 +258,32 @@ UNION
 (2 rows)
 
 !ok
+
+# [CALCITE-6955] PruneEmptyRules does not handle the all attribute of SetOp correctly
+select * from (values (10, 1), (30, 3), (30, 3)) as t (x, y)
+union
+select * from (values (20, 2)) as t (x, y) where x > 30;
++----+---+
+| X  | Y |
++----+---+
+| 10 | 1 |
+| 30 | 3 |
++----+---+
+(2 rows)
+
```

Comment:
We need to verify whether the optimizations related to this PR have taken effect through `!plan`.

Reply:
The corresponding plan is already present in RelOptRulesTest.xml. Would you like me to add an additional one here as well.

Reply:
Yes, this is different. In RelOptRulesTest, we have actively added the plan. For the iq file, we are using the default optimization plan.

Reply:
If this plan doesn't take effect, this SQL statement should still be able to execute successfully.

Reply:
Oops! You're completely right. I'll add it soon.

Reply:
@NobiGo Done!